suffuse/root.go

```diff
@@ -0,0 +1,86 @@
+package suffuse
+
+import (
+  "os"
+  "bazil.org/fuse"
+)
+
+/** An interface encapsulating the essential fuse operations.
+ *  Each method returns nil if inapplicable. The fuse operations
+ *  use the first non-nil return value found for a given Path.
+ */
+type Rule interface {
+  MetaData(Path)*fuse.Attr
+  FileData(Path)[]byte
+  DirData(Path)[]fuse.Dirent
```

Comment:
If this `Rule` interface will stay here when the 'engine' is moved, I think we should abstract over the types from the `fuse` package.